### ISSUE TITLE
Call for reviews & review process

### ISSUE THREAD

**Issue by Beanow** [sfosc/sfosc#71]
As I mentioned in #70:
> I'm happy to see @nothingismagick made a round of several github reviews and propose moving forward with making this the normal workflow.

So with this issue I would like to propose a workflow and kindly ask for anyone willing to help out by submitting reviews.

To make this a concrete proposal, I would suggest:
- No reviewing of your own PRs.
- Going with an open / low requirements membership policy in #64.
- Use a `reviews-wanted` label to mark PRs.
- Aim for a ~2 day review window since the first **member** review.
- When there aren't any objections and there's at least 1 member review, merge it.
- When there are objections, hold off on the merge until there is consensus.
- Move discussions that fail to reach consensus to an issue and close the PR.
- A maintainer can extend the review window and/or raise the required review count if they deem it to be appropriate for more controversial or large changes.
- Anyone is free to raise concerns later if they missed their chance. Issues would be the recommended way to do so.

My thinking is that this will strike a balance between timely merges and a chance for the community to share their thoughts on PRs. I'm suggesting member reviews are treated specially but access to membership is kept broad, to avoid the situation merges will be seen as carrying too little weight in terms of community approval.

What do you think of this approach for SFOSC?

**Comment by nothingismagick** [comment]
I would agree however with the exception of requiring two reviews. I also think that there should be something like a CLA where contributors expressly permit their contribution to be used.

Something like this can be integrated into the PR pipeline:
https://cla-assistant.io/

I've used it before as a github integration and it works fine.

**Comment by Beanow** [comment]
Interesting, I'll look into it. CLA can be worthwhile but definitely would like to dig a bit deeper into that before we do. It does always add an additional hurdle, so I think we should look at why we need it and explain this well in the documentation.

**Comment by Beanow** [comment]
Going with 2 reviews minimum seems reasonable to me. I would put the ~2d time window after the second member review as well then. Changed initial post to reflect. I'll look at this and #64 so far and suggest the text for it in PRs.

**Comment by Beanow** [comment]
Experimenting with this on a few merges, this feels right.

@adamhjk could you kindly help us out with the labels? Don't have the access to do this myself.
Some that I would like to build this workflow:
- `reviews-wanted` (inviting color)
- `needs-consensus` (pending status)
- `moved-to-issue` (to acknowledge and index a useful debate, more than closing the PR does)

**Comment by Beanow** [comment]
I've marked several [reviews-wanted](https://github.com/sfosc/sfosc/labels/reviews-wanted) and [needs-consensus](https://github.com/sfosc/sfosc/labels/needs-consensus). Next I'll have a go at drafting the process explanation.
